WordPress/Classes/Utility/Blogging Reminders/BloggingReminders.swift

```diff
@@ -0,0 +1,157 @@
+import Foundation
+
+/// Main interface for scheduling blogging reminders
+///
+class BloggingReminders {
+
+    enum Schedule {
+        /// No reminder schedule.
+        ///
+        case none
+
+        /// Weekdays reminders
+        ///
+        case weekdays(_ days: [Weekday])
+    }
+
+    enum Weekday: Int {
+        case sunday = 1 // Keep this at 1 to match Apple's `DateComponents`' weekday number.
+        case monday
+        case tuesday
+        case wednesday
+        case thursday
+        case friday
+        case saturday
+
+        /// The default reminder hour.  In the future we may want to replace this constant with a more customizable approach.
+        ///
+        static let defaultHour = 10
+    }
+
+    /// Singleton for use in the App.
+    ///
+    static let `shared` = BloggingReminders()
```

Comment:
I admit that I should read the whole thing before commenting on this 😀, but do we really need a singleton here? Since we are creating a new feature that's not interconnected with existing code, I'd expect we can inject instances of objects without singletons

Reply:
I think you're right on this.  I'll try playing a bit with the code to remove the singletons.

As a side note this made me wonder if we want `BloggingRemindersStore` to be thread safe, since it's writing and reading from a file.

Reply:
Removed the singletons.